### DISCUSSION TITLE
Middleware

### DISCUSSION THREAD

**Discussion by Notadumdum** [miguelgrinberg/python-socketio#1066] [Q&A] [answered]
I wanna add some token for authorization of the client... and I couldn't find anything on middleware in documentation version 4.6.1.. please could you tell me about it

**Comment by miguelgrinberg** [comment, marked as answer]
Client authentication is done in the `connect` handler.

**Reply by Notadumdum** [reply]
The connection keeps on closing so I think I need to pass an extra token that gets sent before the normal authorization if that makes sense

**Reply by Notadumdum** [reply]
![Screenshot_20221016-164431_Kiwi Browser](https://user-images.githubusercontent.com/105162586/196044673-b790aa09-16e6-4d93-a830-736b22f517aa.jpg)

**Reply by miguelgrinberg** [reply]
The Socket.IO protocol does not define how to do authentication, the server can look for authentication details in headers, cookies or the connection URL itself. The newer version of the protocol (the one you are not using) has an explicit way to pass auth information during a connection, but this is optional and then again, you are using an older version, so this does not apply to you.

In short, ask the people who work on the server what you need to send. They are the only ones who know.

**Reply by Notadumdum** [reply]
OK thanks but I have one more question
![Screenshot_20221017-075539_Kiwi Browser](https://user-images.githubusercontent.com/105162586/196109163-fdb9ac16-f901-4f89-9c0f-0055ae7c7ff3.jpg)

If the "upgrades" array is empty after successful connection of the browser.. does it mean anything??
Would it be any clue to me making a connection work

**Reply by miguelgrinberg** [reply]
The Engine.IO protocol is publicly available: https://github.com/socketio/engine.io-protocol. Once you familiarize yourself with Engine.IO you can look at the Socket.IO protocol which builds on top of it here: https://github.com/socketio/socket.io-protocol
